src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVCustomReadFilter.java

```diff
@@ -0,0 +1,94 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * Customized serializable reads filter, based on cmd line arguments provided, for structural variation purposes
+ */
+public final class SVCustomReadFilter implements ReadFilter {
+    private static final long serialVersionUID = 1L;
+
+    private static final PrimaryFilter pfilter = new PrimaryFilter();
+    private final CustomFilter sfilter;
+
+    public SVCustomReadFilter(final boolean[] sixFilterOptions,
+                              final CollectInsertSizeMetricsSpark.whichEndToUse whichEnd,
+                              final int mappingQualityThreshold
+                              //TODO: final int mateMappingQualityThreshold
+                             ){
+        sfilter = new CustomFilter(sixFilterOptions, whichEnd, mappingQualityThreshold);//TODO:, mateMappingQualityThreshold;
+    }
+    @Override
+    public boolean test(final GATKRead read){
+        return pfilter.and(sfilter).test(read);
```

Comment:
create the `pfilter.and(sfilter)` object in the constructor and only call `test` on that object here - otherwise your creating the same object over and over, once for each read